modules/serverless-apiserversource.adoc

```diff
@@ -0,0 +1,231 @@
+ApiServerSource watches for Kubernetes events and forwards them to the Knative Eventing broker.
+
+.Prerequisites
+
+* Using ApiServerSource requires you to have installed Knative Serving and Eventing. For more information, see the documentation on xref:../../serverless/installing-openshift-serverless.adoc#installing-openshift-serverless[Installing OpenShift Serverless].
+
+* You must have created the `default` broker in the same namespace as the one defined in the ApiServerSource YAML. For more information, see the documentation on xref:../../serverless/knative_eventing/serverless-using-brokers.adoc#serverless-using-brokers[Using brokers with Knative Eventing].
+
+* For steps that use `kn`, you will need to have the `kn` CLI installed. For more information, see the documentation on xref:../../serverless/installing-kn.adoc#installing-kn[Installing the Knative CLI].
+
+.Procedure
+
+. Create a Service Account for the ApiServerSource.
++
+You can do this by creating a file named `serviceaccount.yaml` and copying the following sample code into it:
++
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: events-sa
+  namespace: default
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: event-watcher
+rules:
+  - apiGroups:
+      - ""
+    resources:
+      - events
+    verbs:
+      - get
+      - list
+      - watch
+
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-ra-event-watcher
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: event-watcher
+subjects:
+  - kind: ServiceAccount
+    name: events-sa
+    namespace: default
+----
++
+[NOTE]
+====
+If you want to re-use an existing Service Account with the appropriate permissions, you must modify the `serviceaccount.yaml` for that Service Account.
+====
++
+Create the service account by entering the following command:
++
+----
+$ oc apply --filename serviceaccount.yaml
+----
+
+. Create an ApiServerSource event source.
++
+You can do this by creating a file named `k8s-events.yaml` and copying the following sample code into it:
++
+----
+apiVersion: sources.knative.dev/v1alpha1
+kind: ApiServerSource
+metadata:
+  name: testevents
+  namespace: default
+spec:
+  serviceAccountName: events-sa
+  mode: Resource
+  resources:
+    - apiVersion: v1
+      kind: Event
+  sink:
+    ref:
+      apiVersion: eventing.knative.dev/v1beta1
+      kind: Broker
+      name: default
+----
++
+[NOTE]
+====
+If you want to consume events from a different namespace or use a different Service Account, you need to modify the sample `k8s-events.yaml` file accordingly.
+====
++
+Create the event source by entering the following command:
++
+----
+$ oc apply --filename k8s-events.yaml
+----
+
+. To check that the ApiServerSource is set up correctly, create a Knative service that dumps incoming messages to its log.
++
```

Comment:
You can use the following `kn` command todo the same:

```
kn service create event-display --image quay.io/openshift-knative/knative-eventing-sources-event-display
```
which leads to this output

```
Creating service 'event-display' in namespace 'demo':

  0.428s The Route is still working to reflect the latest desired specification.
  0.544s Configuration "event-display" is waiting for a Revision to become ready.
 21.336s ...
 21.629s Ingress has not yet been reconciled.
 21.847s Ready to serve.

Service 'event-display' created to latest revision 'event-display-qlnmh-1' is available at URL:
http://event-display.demo.example.com
```

(the URL displayed will different on openshift though)